clang/test/CIR/CodeGen/union-array.c

```diff
@@ -37,3 +39,23 @@ void bar(void) {
 }
 // CIR: cir.global "private" internal dsolocal @g = #cir.const_record<{#cir.int<5> : !s32i}> : !rec_anon_struct
 // CIR: cir.const #cir.const_array<[#cir.global_view<@g> : !cir.ptr<!s32i>, #cir.global_view<@g> : !cir.ptr<!s32i>]> : !cir.array<!cir.ptr<!s32i> x 2>
+
+typedef struct {
+    long s0;
+    int  s1;
+} S_3;
+
+typedef union {
+   int  f0;
+   S_3 f1;
+} U2;
+
+
+static U2 g1[3] = {{0x42},{0x42},{0x42}};
+int* g2 = &g1[1].f1.s1;
+// CIR: cir.global external @g2 = #cir.global_view<@g1, offset 24> : !cir.ptr<!s32i>
```

Comment:
Shouldn't this test plain `LLVM` as well?

Reply:
I just wanted to show how the original LLVM looks like

Reply:
OG is fine, just asking for the extra one too